### PR TITLE
feat(nacl): add sealBox and unsealBox to SKP

### PR DESCRIPTION
**Pull request by chunningham**
implemented, awaiting vkp interface definition changes in jolo-proto-ts

### CODE REVIEW

File: tests/data/didDocument.data.ts [hunk 1]
```diff
@@ -37,6 +39,40 @@ export const didDocumentJSON = {
...
+  publicKey: [
+    {
+      id: mockKeyId,
+      type: 'Secp256k1VerificationKey2018',
+      owner: mockDid,
+      publicKeyHex: mockPublicKeyHex,
+    },
+    {
+      id: mockEncKeyId,
+      type: 'X25519KeyAgreementKey2019',
```
Comment (Exulansis) [inline tests/data/didDocument.data.ts:58]: According to the definition here - (context)[https://identity.foundation/context/security.html#X25519KeyAgreementKey2019]

This is deprecated, and should not be used. 
An alternative I've seen used is `Curve25519EncryptionPublicKey`, but it's not listed in any contexts as far as I can tell...

Comment (Exulansis) [reply tests/data/didDocument.data.ts:58]: Actually nvm, the [did-core registry](https://github.com/w3c/did-spec-registries) only lists `X25519KeyAgreementKey2019` with a note _"Normative definition pending"_, so I assume this is the right type to use.

File: tests/data/didDocument.data.ts [hunk 2]
```diff
@@ -37,6 +39,40 @@ export const didDocumentJSON = {
...
+      owner: mockDid,
+      publicKeyHex: mockEncKeyHex,
```
Comment (Exulansis) [inline tests/data/didDocument.data.ts:60]: All examples of `X25519KeyAgreementKey2019` use `publicKeyBase58`. Not sure if this matters though.

File: ts/vaultedKeyProvider/softwareProvider.ts [hunk 3]
```diff
@@ -117,15 +119,27 @@ export class SoftwareKeyProvider implements IVaultedKeyProvider {
    * @example `vault.getPublicKey({derivationPath: ..., decryptionPass: ...}) // Buffer <...>`
    */
 
-  public getPublicKey(derivationArgs: IKeyDerivationArgs): Buffer {
+  public getPublicKey(
+    derivationArgs: IKeyDerivationArgs,
+    scheme = SchemeTypes.secp256k1,
```
Comment (Exulansis) [inline ts/vaultedKeyProvider/softwareProvider.ts:124]: Should this be part of `derivationArgs`?

File: ts/vaultedKeyProvider/softwareProvider.ts [hunk 4]
```diff
@@ -399,3 +457,17 @@ export class SoftwareKeyProvider implements IVaultedKeyProvider {
     return passwordBuffer
   }
 }
+
+// Clamps a Curve25519 private key to prevent a few key attacks
+const normalizeX25519PrivKey = (key: Buffer): Buffer => {
+  // clamp the lower bits, ensuring the key is a multiple of the cofactor, preventing small subgroup attacks
+  key[0] &= 248
+
+  // clamp the second most upper bit (something to do with montgomery ladder implementations)
```
Comment (Exulansis) [inline ts/vaultedKeyProvider/softwareProvider.ts:466]: :D

File: ts/registries/jolocomRegistry.ts [hunk 5]
```diff
@@ -54,9 +55,23 @@ export class JolocomRegistry implements IRegistry {
       encryptionPass: decryptionPassword,
     }
 
+    const encArgs = {
+      derivationPath: "m/74'/0'/0'/0",
```
Comment (Exulansis) [inline ts/registries/jolocomRegistry.ts:59]: Should this be defined somewhere else?

File: tests/jolocomRegistry/create.test.ts [hunk 6]
```diff
@@ -42,7 +45,8 @@ describe('Jolocom Registry - create', () => {
   it('should create new identity', async () => {
     const jolocomRegistry = createJolocomRegistry()
 
-    const expectedDidDoc = Object.assign({}, didDocumentJSON)
+    const pkDidDoc = Object.assign({}, didDocumentJSON)
```
Comment (Exulansis) [inline tests/jolocomRegistry/create.test.ts:48]: Is this needed?

File: ts/vaultedKeyProvider/softwareProvider.ts (hunk 3 above)
Comment (chunningham) [reply ts/vaultedKeyProvider/softwareProvider.ts:124]: maybe that would be better, this just avoids a breaking change

File: tests/data/didDocument.data.ts (hunk 2 above)
Comment (chunningham) [reply tests/data/didDocument.data.ts:60]: nah the security vocab leaves it up to the encoder

**Pull request merged**